packages/backend/src/open_payments/payment_pointer/model.test.ts

```diff
@@ -353,4 +366,61 @@ export const getRouteTests = <M extends PaymentPointerSubresource>({
   }
 }
 
-test.todo('test suite must contain at least one test')
+describe('Payment Pointer Model', (): void => {
+  let deps: IocContract<AppServices>
+  let appContainer: TestContainer
+
+  beforeAll(async (): Promise<void> => {
+    deps = initIocContainer(Config)
+    appContainer = await createTestApp(deps)
+  })
+
+  afterEach(async (): Promise<void> => {
+    await truncateTables(appContainer.knex)
+  })
+
+  afterAll(async (): Promise<void> => {
+    await appContainer.shutdown()
+  })
+
+  describe(`deactivatedAt cases @ ${new Date().toISOString()}`, () => {
```

Comment:
```suggestion
  describe(`deactivatesAt cases @ ${new Date().toISOString()}`, () => {
```

Reply:
I would go towards using `deactivatedAt` for everything IMO, since `deactivatesAt` seems like it should be used for the future (ie, for scheduled jobs or something)

Reply:
I'm not sure if we need to have the `{new Date().toISOString()` here, I think just `describes("isActive")` is sufficient since that is what you are testing

Reply:
> I would go towards using deactivatedAt for everything IMO, since deactivatesAt seems like it should be used for the future (ie, for scheduled jobs or something)

Currently the timestamp will always be in the past after its set. But should it always? Currently there is nothing preventing it from being set to a future date which is why I went with `deactivatesAt`. 

I realize this isnt a current use-case but If we wanted to we could set a `deactivatesAt`to a future value and then when that time comes it will be deactivated. I feel like `deactivatesAt` is truer to what the value could be. If it should not be a value in the future I wonder if I should add something to the model preventing it, because obviously the database will accept any timestamp.

> I'm not sure if we need to have the {new Date().toISOString() here, I think just describes("isActive") is sufficient since that is what you are testing

I included the date in the describe name because otherwise its not clear from the test names whether `deactivatesAt` being tested is in the future or the past: 

```
  Payment Pointer Model
    deactivatedAt cases @ 2023-05-11T12:08:21.637Z
      ✓ For a deactivatesAt of null, status should be ACTIVE and isActive should be true (109 ms)
      ✓ For a deactivatesAt of 2023-05-12T12:08:21.637Z, status should be ACTIVE and isActive should be true (70 ms)
      ✓ For a deactivatesAt of 2023-05-10T12:08:21.637Z, status should be INACTIVE and isActive should be false (97 ms)
```

Although now I'm starting to think I could just set an explicit name for each case like "Future deactivatesAt isActive should be true" and "Past deactivatesAt isActive should be false".

Reply:
I actually liked the forward thinking here and that you could deactivate a payment pointer at a given date and time in the future. That's what I can do with my bank account, too.

Reply:
I think i'm changing my mind after realizing setting a future date only works out-of-the-box for some cases. It works when we read payment pointers then doing something based on its active/inactive state. It wont work if we want to perform some action at the time it becomes invalid - we'd need to build something additional and it probably wouldn't be driven by a future `deactivatesAt` field.

Consider the possible direction @sabineschaller's offered for dealing with invalid payment pointers and incoming payemnts: 

> Maybe we should complete all open incoming payments when a payment pointer is deactivated.

With respect to setting a future date on `deactivatesAt`, that would require something to constantly watch and do something with dates that switch from being in the future to in the past. But this is probably not the way we'd want to do it. Instead I imagine we'd schedule some event to happen in the future, at which point a `deactivatedAt` field goes from `null` to the current time.

Reply:
> It works when we read payment pointers then doing something based on its active/inactive state. It wont work if we want to perform some action at the time it becomes invalid - we'd need to build something additional and it probably wouldn't be driven by a future deactivatesAt field.

I agree. Until we do some sort of scheduling (that will require adding addtional flags, table, etc), `deactivatedAt` field is the simplest solution at the moment. I don't think we need to validate this at the DB layer either IMO, as long as the application layer is setting `deactivatedAt = now` as it is doing now.

Reply:
i renamed back to `deactivatedAt`